src/url.rs

```diff
@@ -5,14 +5,20 @@ use crate::scanner::Scanner;
 /// Scan for URLs starting from the trigger character ":", requires "://".
 ///
 /// Based on RFC 3986.
-pub struct UrlScanner {}
+pub struct UrlScanner {
+    pub trigger: char,
+}
 
 impl Scanner for UrlScanner {
-    fn scan(&self, s: &str, colon: usize) -> Option<Range<usize>> {
-        let after_slash_slash = colon + 3;
+    fn scan(&self, s: &str, trigger_index: usize) -> Option<Range<usize>> {
+        let (proto, offset) = match self.trigger {
+            '/' => ("//", 2),
+            _ => ("://", 3),
```

Comment:
If you like, we can make the trigger an `enum` type. Then we can use an exhaustive match here:

```rust
let (proto, offset) = match self.trigger {
  Trigger::Slash => ("//", 2),
  Trigger::Colon => ("://", 3),
}
```

Reply:
Ah I just went ahead and added it. Think that's cleaner now. Let me know if I should revert the change. ✌️